interfaces/rerank/README.md

```diff
@@ -0,0 +1,64 @@
+`@supplies` directive connects a field selection on a concrete type to an interface request. 
+
+For example, you can use interface queries when you want to be able to run the same field request against multiple backends.
+
+This same feature can also be used to provide alternative logic to execute when a field is selected.
+
+In this snippet we introduce a very generic interface `JSONDocs` that has a single field to contain a `JSON` field `docs`, and a single implementation of that interface `Concrete`.
+
+```
+interface JSONDocs {
+    docs: JSON
+}
+
+type Concrete implements JSONDocs {
+    docs: JSON
+}
+```
+
+This snippet demonstrates the alternative execution paths, with `reRank` field selection being *supplied* by the fields `reRankAsc`, `reRankDesc`, and `reRankNoop`.
```

Comment:
Only issue is that the reader is left on their own to figure out how any of this is done. Some more explanation in the comments would have helped, e.g. point the reader towards the 'if' argument in the supplies and how it is used.